File: URI/C/notasemoedas.c
#include <stdio.h>
#include <math.h>


int main()
{
    float dinheiro, aux;
    int notasv[]  = {100, 50, 20, 10, 5, 2};
    int moedasv[] = {100, 50, 25, 10, 5, 1};
    float moedasAUX[] = {1.0, 0.50, 0.25, 0.10, 0.05, 0.01};
    scanf("%f", &dinheiro);

    int parteInteira = (int)dinheiro;
    int parteFracionada = round((dinheiro - parteInteira)*100);

    puts("NOTAS:");
    for (int i = 0; i < 6; i++)
    {
        if (parteInteira >= notasv[i])
        {
            aux = parteInteira;
            parteInteira = parteInteira / notasv[i];
            printf("%d nota(s) de R$ %i.00\n",parteInteira, notasv[i]);
            parteInteira = aux - parteInteira*notasv[i];
        }
        else
        {
            printf("0 nota(s) de R$ %i.00\n", notasv[i]);
        }
    }
    parteFracionada+=parteInteira*100;
    puts("MOEDAS:");
    for (int i = 0; i < 6; i++)
    {
        if (parteFracionada >= moedasv[i])
        {
            aux = parteFracionada;
            parteFracionada = parteFracionada / moedasv[i];
            printf("%d moeda(s) de R$ %.2f\n",parteFracionada, moedasAUX[i]);
            parteFracionada = aux - parteFracionada*moedasv[i];
        }
        else
        {
            printf("0 moeda(s) de R$ %.2f\n", moedasAUX[i]);
        }
    }
    

    return 0;
}

File: URI/Java/NotasEMoedas.java
package Java;

import java.util.Scanner;

public class NotasEMoedas {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        float din = input.nextFloat();
        int pi = (int)din;
        int pd = (int)(din*100 - pi*100);

        System.out.println("NOTAS:");
        System.out.println(pi/100 + " nota(s) de R$ 100.00");
        pi = pi%100;
        System.out.println(pi/50 + " nota(s) de R$ 50.00");
        pi = pi%50;
        System.out.println(pi/20 + " nota(s) de R$ 20.00");
        pi = pi%20;
        System.out.println(pi/10 + " nota(s) de R$ 10.00");
        pi = pi%10;
        System.out.println(pi/5 + " nota(s) de R$ 5.00");
        pi = pi%5;
        System.out.println(pi/2 + " nota(s) de R$ 2.00");
        pi = pi%2;
        pd+= 100*pi;
        System.out.println("MOEDAS:");
        System.out.println(pd/100 + " moeda(s) de R$ 1.00");
        pd %= 100;
        System.out.println(pd/50 + " moeda(s) de R$ 0.50");
        pd %= 50;
        System.out.println(pd/25 + " moeda(s) de R$ 0.25");
        pd %= 25;
        System.out.println(pd/10 + " moeda(s) de R$ 0.10");
        pd %= 10;
        System.out.println(pd/5 + " moeda(s) de R$ 0.05");
        pd %= 5;
        System.out.println(pd + " moeda(s) de R$ 0.01");
        
        input.close();
    }
}